packages/microsoft-reactnative-sampleapps/windows/SampleAppCPP/App.cpp

```diff
@@ -34,6 +34,13 @@ App::App() noexcept {
   InstanceSettings().EnableDeveloperMenu(false);
 #endif
 
+#if USE_DIRECT_DEBUGGING
+  InstanceSettings().UseWebDebugger(false);
+  InstanceSettings().UseDirectDebugger(true);
+  InstanceSettings().DebuggerBreakOnNextLine(true);
+  InstanceSettings().DebuggerPort(9229);
```

Comment:
these things can just go into EnableDeveloperMenu(true) actually?

Is direct debugging mutually exclusive with regular web debugging? If not, let's just turn it on for debug builds if it's overall goodness

Reply:
They are mutually exclusive.

Reply:
@acoates-ms can it be made non-mututally exclusive? is it a matter of just having two ports open?

Reply:
No.  Web Debugging means the JS runs outside of the app.  Direct debugging enables a port on the JS engine within the app.

Reply:
Some context:  
Web debugging is the original debugging feature.  The idea is that the javascript portion of your app is running in a web browser so that you can use the browser's debugging tools to debug.  The "bridge" establishes a bi-directional communication channel between the native code and the javascript (using websocket).

Direct debugging allows you to debug your javascript while running directly in your client process (like it does in a production environment).  So the two are mutually exclusive by design.
 
Web debugging is going to become a thing of the past as the platform evolves, due to the introduction of JSI, which allows JS to commumicate directly with native code.  This is the basis for turbo modules, and eventually fabric.  In this world, web debugging won't work, as there's no bridge.

Reply:
How do we feel about adding a new template where the debug builds use direct debugging instead of web debugging?

Reply:
I'm not a fan of that idea.  You would not be able to switch between direct debugging/web debugging without changing code, which you can already do just by changing some code.  In my opinion the goal of this work should be to allow the user to choose what type of debugging they want at run-time.

Andrew is raising a good point that the template app has logic that other apps might not have, so if we add a CLI parameter, we should find a way to make that parameter work in all scenarios.  This likely requires re-working how we're pushing all debugging settings into the core.  I personally would be fine with using the #define in the template as a starting point and file a separate issue to tackle this.

Reply:
Thank you for the feedback @acoates-ms @asklar @kmelmon.
I replaced the defines with command line arguments. To my understanding with the new approach:
- You can toggle between direct debugging and web debugging without rebuilding.
- Is compatible with existing projects (is not app specific).
- You can also specify the port (C# defines can't be assigned values, so previous approach didn't allow that).
Unfortunately there still isn't an equivalent option in the android and iOS CLI.
Let me know if you have any concerns.